code/datums/client_image_controller.dm

```diff
@@ -0,0 +1,359 @@
+		Adds a client image to the system
+		"is_unique_image = TRUE" is necessary when an image is going to be stored in multiple key groups.
+
+		i.e.)
+			GLOB.cimg_controller.stack_client_images(KEY_CULT, a_cult_image, is_shared_image = TRUE)
+			GLOB.cimg_controller.stack_client_images(KEY_CLICK_CULT, a_cult_image, is_shared_image = TRUE)
+			GLOB.cimg_controller.stack_client_images(KEY_CHAPLAIN, a_cult_image, is_shared_image = TRUE)
+
+		a_cult_image will be given to a client multiple times when they have KEY_CULT, KEY_CLICK_CULT, KEY_CHAPLAIN at the same time
+		This is why "is_shared_image = TRUE" is necessary
+
+	* GLOB.cimg_controller.cut_client_images(cimg_key, client_images, is_shared_image = FALSE)
+		Removes a client image from the system
+		! Do not forget "var/image/image = null"
+		! Do not forget to set "is_shared_image = TRUE" if the image was given with "is_shared_image = TRUE"
+
+	* GLOB.cimg_controller.nuke_client_image_holder(cimg_key)
+		Removes a cimg holder entirely from the system.
+		If things are common (like holy tiles), this should be avoided
+		If things are only dedicated to a single person, you may use this
+
+	------------------------
+	[To mobs/mind that need to see client images]
+	* GLOB.cimg_controller.validate_mob(cimg_key, mob/cimg_mob)
+	* GLOB.cimg_controller.validate_mind(cimg_key, datum/mind/cimg_mind)
+		Give a mob/mind an ability to see images in "cimg_key" group
+		Validation is stackable, so be careful if you're going to grant it multiple times
+		Both(mob/mind) does the same thing, but mob/mind difference.
+
+	* GLOB.cimg_controller.disqualify_mob(cimg_key, mob/cimg_mob)
+	* GLOB.cimg_controller.disqualify_mind(cimg_key, datum/mind/cimg_mind)
+		Removes a mob/mind's ability to see images in "cimg_key" group
+		Validation is stackable, so they'll be still capable of seeing stuff when validation is given multiple times.
+		Both(mob/mind) does the same thing, but mob/mind difference.
+
+
+		< What's best to use between mob and mind? >
+			GLOB.cimg_controller.validate_mind("heretic_pierces", heretic_mind)
+		You want heretics have this always even if their body is changed (likely brain transplant)
+		aghost will still be capable of seeing stuff, because their mind is validated and ghost has that mind.
+
+			GLOB.cimg_controller.validate_mob("heretic_pierces", target_mob)
+		If a curator wears goggles that can see "heretic_pierces", we want to use this proc instead of mind.
+		aghost will rip off the images because they become a ghost, but their aghost is not validated.
+		once aghost goes back to their mob, they'll see the stuff again.
+*/
+
+/datum/cimg_controller
+	/// list of /datum/cimg_holder datum types
+	var/list/cimg_holders = list()
+
+	/// a quick key list for on mob login
+	var/list/cimgkey_by_mob = list()
+	/// a quick key list for on mob login (mob login checks mind too)
+	var/list/cimgkey_by_mind = list()
```

Comment:
You should use PRIVATE_VAR to ensure that these aren't being accessed from outside